components/net/websocket_loader.rs

```diff
@@ -368,16 +373,37 @@ fn connect(
         .read()
         .unwrap()
         .apply_hsts_rules(&mut req_builder.url);
+    let request = req_builder.build();
 
-    let req_url = req_builder.url.clone();
+    let req_url = request.url();
+    let req_origin = match request.origin {
+        Origin::Client => unreachable!(),
+        Origin::Origin(ref origin) => origin,
+    };
 
     if should_request_be_blocked_due_to_a_bad_port(&req_url) {
         return Err("Port blocked".to_string());
     }
 
+    let policy_container = match &request.policy_container {
+        RequestPolicyContainer::Client => PolicyContainer::default(),
```

Comment:
Can the policy container actually be `"client"` at this point?
I think it's initialized fairly early on in the fetch machinery.

Reply:
I will let Josh respond to this, as he wrote it 😛

Reply:
Right now this use is separate from the one in the fetch implementation (#35028) so I'd prefer to keep this.